client/context/query.go

```diff
@@ -81,6 +81,19 @@ func (ctx CLIContext) query(path string, key cmn.HexBytes) (res []byte, height i
 		return res, height, err
 	}
 
+	// When a client did not provide a query height, manually query for it so it can
+	// be injected downstream into responses.
+	if ctx.Height == 0 {
+		status, err := node.Status()
+		if err != nil {
+			return res, height, err
+		}
+		if status.SyncInfo.CatchingUp {
```

Comment:
Just to clarify... if i query against a syncing node, I have to provide a height along with query or it will error, correct?

Reply:
Queries should error against a syncing node IMO.

Reply:
Yea that makes sense to me. I would add a comment here for clarity

Reply:
I don't even think this is necessary. You should still be able to query against that is syncing, albeit you should know it's syncing and not even bother querying it.